local/ob-tmate/ob-tmate.el

```diff
@@ -245,22 +242,27 @@ Argument OB-SESSION: the current ob-tmate session.
 Optional command-line arguments can be passed in ARGS."
   (message "OBSESSION: %S" ob-session)
   (message "OBSESSION ARGS: %S" args)
-  (if (ob-tmate--socket ob-session)
-      (progn
-        (message (concat "OB-TMATE: execute on provided socket: => " (ob-tmate--socket ob-session)))
-        (message (concat "OB-TMATE: execute args: => " (string-join args " ")))
-        (message (concat "OB-TMATE: applying 'start-process"))
-        (setenv "TMUX") ;unset tmux env so this can be run from within a tmux session without complaint
-        (apply 'start-process "ob-tmate" "*Messages*"
-	             org-babel-tmate-location
-	             "-S" (ob-tmate--socket ob-session)
-	             args))
-    (progn
-      (message (concat "OB-TMATE: execute args: => " (string-join args " ")))
-      (message (concat "OB-TMATE: execute start-process:" (ob-tmate--socket ob-session)))
-      (setenv "TMUX")
-      (apply 'start-process
-	           "ob-tmate" "*Messages*" org-babel-tmate-location args))))
+  (let ((process-environment (cl-copy-list process-environment)))
+       (setenv-internal process-environment "TMUX" nil nil) ; because of the let, process-environment is local
+       ;; unset tmux env so this can be run from within a tmux session without complaint
```

Comment:
`setenv-internal` allows us to specific precisely which (local) variable we want to set.

In this case we /set/ `TMUX` to nil... effectively clearing it.